### ISSUE TITLE
Let autodiff support nn like resnet

### ISSUE THREAD

**Issue by waynehpc** [cryinlaugh/SWCompiler#4]
Support Training of Resnet may be a must, or we could only supported layered model now, that's not persuasive.

In resnet or inception-like net, one TensorNode may have multiple out-degree, which means there are multiple gradients of this tensornode during backpropogation. 
![image](https://user-images.githubusercontent.com/44743868/65398559-87af2900-ddea-11e9-873c-4bebde32a629.png)
https://github.com/cryinlaugh/SWCompiler/blob/f9c13382c550e71e21220ef3f1b6d5ff4a8702d9/src/op/dlOp/dlOpAD.cpp#L203-L204

The fix may be easy: before updating map, if gradient already exists, then create an ElementAdd OpNode, and sum them. 

But we need sometime to **fix** and **test** this.

**Comment by paradoxstar** [comment]
I will finish the network(resnet) describe，graph generation including all passes in next two days.
